zio-http/src/main/scala/zhttp/http/Scheme.scala

```diff
@@ -1,29 +1,70 @@
 package zhttp.http
 
 import io.netty.handler.codec.http.HttpScheme
+import io.netty.handler.codec.http.websocketx.WebSocketScheme
+import zhttp.http.Scheme.{HTTP, HTTPS, WS, WSS}
+
 sealed trait Scheme { self =>
-  def encode: String = Scheme.asString(self)
-}
-object Scheme       {
-  def asString(self: Scheme): String = self match {
+  def encode: String = self match {
     case HTTP  => "http"
     case HTTPS => "https"
+    case WS    => "ws"
+    case WSS   => "wss"
+  }
+
+  def isHttp: Boolean = !isWebSocket
+
+  def isWebSocket: Boolean = self match {
+    case Scheme.WS  => true
+    case Scheme.WSS => true
+    case _          => false
+  }
+
+  def isSecure: Boolean = self match {
+    case Scheme.HTTPS => true
+    case Scheme.WSS   => true
+    case _            => false
+  }
+
+  def toJHttpScheme: Option[HttpScheme] = self match {
+    case HTTP  => Option(HttpScheme.HTTP)
+    case HTTPS => Option(HttpScheme.HTTPS)
+    case _     => None
+  }
+
+  def toWebSocketScheme: Option[WebSocketScheme] = self match {
```

Comment:
```suggestion
  def toJWebSocketScheme: Option[WebSocketScheme] = self match {
```
for consistency in nomenclature

Reply:
This is done. Thanks, Amit.